src/main/java/com/jacob/duke/Storage.java

```diff
@@ -0,0 +1,228 @@
+        accessTaskListInFileSystem(getCurrentDirectory());
+    }
+
+    //get current directory
+    private String getCurrentDirectory() {
+        return System.getProperty("user.dir");
+    }
+
+    //Access the current list, creates the folder and files if they do not exist
+    private void accessTaskListInFileSystem(String current) {
+        String[] fileParents = filename.split("/");
+        String parent = fileParents[0];
+        java.nio.file.Path directoryPath = java.nio.file.Paths.get(current, parent);
+        boolean directoryExists = java.nio.file.Files.exists(directoryPath);
+
+
+        java.nio.file.Path filePath = java.nio.file.Paths.get(current, filename);
+        boolean fileExists = java.nio.file.Files.exists(filePath);
+        try {
+            if (!directoryExists) {
+                Files.createDirectories(directoryPath);
+            }
+            if (!fileExists) {
+                Files.createFile(filePath);
+            }
+        } catch (IOException e) {
+            System.out.println(e.getMessage());
+        }
+        //System.out.println("Printing filepath after accessing it: " + filePath.toString());
+    }
+
+
+    //handle the file lines at initialization
+    private void handleFileCommands(String inputCommand, List<Task> taskList, int count) {
+
+        //parse input command (in format :  type, done, description, datetime) into task
+        String[] inputs = inputCommand.split(",");
+        String type = inputs[0];
+        int isDone = Integer.parseInt(inputs[1]);
+        String description = inputs[2];
+
+        switch (type) {
+        case "T":
+            Task theTodo = new Todo(description);
+            if (isDone == 1) {
+                theTodo.setDone();
+            }
+            taskList.add(count, theTodo);
+            break;
+        case "E":
+            String dateTime = inputs[3];
+            Task theEvent = new Event(description, dateTime);
+            if (isDone == 1) {
+                theEvent.setDone();
+            }
+            taskList.add(count, theEvent);
+            break;
+        case "D":
+            dateTime = inputs[3];
+            Task theDeadline = new Deadline(description, dateTime);
+            if (isDone == 1) {
+                theDeadline.setDone();
+            }
+            taskList.add(count, theDeadline);
+            break;
+        default:
+            break;
+        }
+    }
+
+    /**
+     * Handles the file and creates the taskList objects based on file information
+     * @return List which containing the task objects in a list representation
+     */
+    public List<Task> readFile() {
+        List<Task> taskList = new ArrayList<>();
+        try {
+            BufferedReader reader = new BufferedReader(new InputStreamReader(
+                    new DataInputStream(new FileInputStream(this.filename))));
+            int count = 0;
+            for (String line; (line = reader.readLine()) != null; ) {
```

Comment:
Interesting way of using a for loop, I'm not very sure how it works but if (String) line is not updated then maybe you can use a while loop instead?